gap/attr.gi

```diff
@@ -128,18 +135,29 @@ function(digraph)
     if not IsNullDigraph(subdigraph)
         and not IsBipartiteDigraph(subdigraph) then
       if IsCompleteDigraph(DigraphSymmetricClosure(subdigraph)) then
-        if n > chrom then
+        if chrom = fail or n > chrom then
           chrom := n;
         fi;
       else
-        for i in [3 .. n - 1] do
-          if DigraphColoring(subdigraph, i) <> fail then
-            if i > chrom then
-              chrom := i;
+        upper := RankOfTransformation(DigraphColouring(subdigraph), nr);
+        if chrom = fail or upper > chrom then
+          upper := Minimum(upper, nr - 1);
+          lower := Maximum(chrom, CliqueNumber(subdigraph));
```

Comment:
If `chrom` is still `fail` at this point, then since `fail` is greater than any integer, this will set `lower = fail`, which you don't want.